features/feature_case/driverMem/driverMem.cu

```diff
@@ -147,7 +159,43 @@ int main(){
     cuMemHostRegister((void *)pFlags, size, CU_MEMHOSTREGISTER_PORTABLE);
 
     cuMemHostUnregister((void *)pFlags);
+}
 
-    return 0;
+int test2() {
+  int ret = 0;
+  std::vector<int> v1, v2;
+  CUdeviceptr p1;
+  CUdeviceptr p2;
+  CUdeviceptr q;
+  auto check = [&](std::string fail) {
+    if (!std::equal(v1.begin(), v1.end(), v2.begin())) {
+      std::cout << fail << "\n";
+      ret = 1;
+    }
+  };
+
```

Comment:
Can we put these tests in a loop that iterates over increasing vector lengths?  Right now the testcase only tests a vector length of 3.